src/services/offerService.js

```diff
@@ -248,13 +265,17 @@ export const newOffer = measureTime(TIMED_ACTIONS.OFFER_CREATE, async ({
         return json;
 
     } catch (error) {
-        if (!isErrorRegistered) createEvent(EVENT_TYPES.ERROR(
+
+        const errorArray = Array.isArray(error) ? error :
+            [{ msg: "Unexpected Error. Please try again later." }];
+
+        if (!isErrorRegistered) createErrorEvent(
```

Comment:
when the function call spans over multiple lines, it's better to have the `{}` around the if body